tools/wake-format/formatter.h

```diff
@@ -59,6 +63,42 @@ struct ctx_t {
   }
 };
 
+struct token_traits_t {
+  // Tokens bound to this token 'before' this token
+  // in source order
+  std::set<CSTElement, CSTElementCompare> before_bound = {};
+
+  // Tokens bound to this token 'after' this token
+  // in source order
+  std::set<CSTElement, CSTElementCompare> after_bound = {};
+
+  // The token this token is bound to
+  // inverse of before/after_bound
+  CSTElement bound_to;
+
+  void bind_before(CSTElement e) {
+    // TODO: delete this after handling captured
+    // NLs and WSes
+    if (e.id() != TOKEN_COMMENT) {
+      return;
+    }
```

Comment:
even when we handle this I think bind_comment will still be clean. The place to handle this will be during emission which will be something like

1) skip the leading/trailing NL
2) skip any WS
3) emit all comments and interior NLs

Reply:
Yeah sorry, I reread that. I know that bind_comment was cleaned up because of the new perspective (that's why I suggested it!) I just mean we have a bit of a hack and it has no home right now. Somewhere has to handle this.